packages/core/src/controllers/subscriber.ts

```diff
@@ -228,13 +237,25 @@ export class Subscriber extends ISubscriber {
     this.logger.debug(`Outgoing Relay Payload`);
     this.logger.trace({ type: "payload", direction: "outgoing", request });
     try {
+      const subId = hashMessage(topic + this.clientId);
+      // in link mode, allow the app to update its network state (i.e. active airplane mode) with small delay before attempting to subscribe
+      if (transportType === TRANSPORT_TYPES.link_mode) {
+        setTimeout(() => {
+          if (this.relayer.connected || this.relayer.connecting) {
+            this.relayer.request(request).catch((e) => this.logger.warn(e));
+          }
+        }, toMiliseconds(ONE_SECOND));
+        return subId;
+      }
```

Comment:
>Wouldn't this add a 1 second delay in every case (even if airplane mode is off).

with link mode, yes but it doesn't cause any issues as we need sometime for the `react-native netinfo` to register network changes when the app is back in foreground 

>On another note we should clear the timeout.

why? there isn't a case where the timeout has to be stopped mid execution

Reply:
> with link mode, yes but it doesn't cause any issues as we need sometime for the react-native netinfo to register network changes when the app is back in foreground

Ideally we should detect when this operation is needed instead of having the delay in every situation but I understand it's probably not trivial.

> why? there isn't a case where the timeout has to be stopped mid execution

It's just a good practice I've been used to because of React development.

From Chat GPT
> Using clearTimeout is not strictly necessary every time you use setTimeout, but there are specific situations where it is beneficial or required:
> 
> Preventing Unwanted Execution: If there's a possibility that the action scheduled with setTimeout should not occur in certain conditions (for example, if the user navigates away from a page or a component is unmounted), then you should clear the timeout to prevent the action from executing when it's no longer needed or relevant.
> 
> Memory Leaks: In React components or other dynamic UI environments, not clearing timeouts can lead to memory leaks. If a timeout is set on a component and the component is unmounted before the timeout fires, the callback function may still be executed, which can lead to errors or unintended behavior, especially if the callback modifies component state.
> 
> Resource Management: Timers hold onto resources until they execute. Clearing them when no longer needed frees up these resources, which is beneficial for performance, especially in environments with limited resources like mobile devices.
> 
> Control Over Asynchronous Actions: In complex applications where multiple timeouts might be set and their behavior could overlap or interfere with each other, managing timeouts with clearTimeout ensures that only the desired timers are running at any given time.